### PR TITLE
 perf(breadcurmb、router): add global breadcrumb and modified router

### PR DESCRIPTION
**Pull request by orangegzx**

### CODE REVIEW

File: src/components/Breadcrumb.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+   * fixed 返回上级面包屑且路由含有动态参数时找不到路由的问题
+   * 1. 进入页面时，标记 & 保存当前路由的fullpath（包含了路由参数）
+   * 2. 存储上一次匹配的路由，即进入页面前的面包屑
+   * 3. 判断当前路由是否是同父级：
+   *  3.1 同父级下，进去嵌套路由（当前路由 matched 的 length > 上一次路由matched.length）：
+   *      3.11 进入嵌套路由：当前理由标记fullpath，便于下一层返回定位，使用上一次matched + 当前路由
+   *      3.12 返回上一层：在上一次的matched 中定位当前路由位置，截取当前路由之前的数据（包含当前路由）
+   *  3.2 不同父级，直接使用新matched
+   * 对比上一次面包屑列表，如当前路由匹配的列表长度比较大，即进入新页面，则合并 2 步骤的列表 和 当前路由（1已记录fullpath）；否则为返回面包屑上一级，则使用当前页面匹配到的列表
+   */
```
Comment (ysfscream) [inline src/components/Breadcrumb.vue:56]: Comments are too long and not in Chinese

Comment (orangegzx) [reply src/components/Breadcrumb.vue:56]: sorry~，forget to remove~

**Comment by orangegzx** [comment]
@ysfscream already

File: src/components/Breadcrumb.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      // go back
+      const i = formMatched.findIndex((item) => item.name === current_route.name)
+      state.levelList = formMatched.slice(0, i + 1)
+    }
+  } else {
+    state.levelList = new_matched
+  }
+}
+
+const onHandleLink = (item:any) => {
```
Comment (ysfscream) [inline src/components/Breadcrumb.vue:70]: I think this item may still type definable, no any.

**Comment by ysfscream** [comment]
LGTM. I merge first

**Review by ysfscream — APPROVED** [review]

**Pull request merged**